linetools/isgm/tests/test_use_abscomp.py

```diff
@@ -252,6 +270,37 @@ def test_repr_alis():
         s = abscomp.repr_alis(fix_strs=('1','2','3','4','5'))
 
 
+def test_repr_joevp():
+    # test with b=0, should be replaced by b_default
+    abscomp, HIlines = mk_comp('HI', b=0*u.km/u.s, use_rand=False)
+    s = abscomp.repr_joevp('test.fits', b_default=3.3*u.km/u.s)
+    assert s == 'test.fits|1215.67|2.92939|13.3|3.3|0.|1|1|1|-300.0|300.0|4772.06378216|4781.6240839|HI\n' \
+    'test.fits|1025.7222|2.92939|13.3|3.3|0.|1|1|1|-300.0|300.0|4026.43131868|4034.49782829|HI\n'
+    # test with b != 0
+    abscomp, HIlines = mk_comp('HI', b=15*u.km/u.s, use_rand=False)
+    s = abscomp.repr_joevp('test.fits', b_default=3.3*u.km/u.s)
+    assert s == 'test.fits|1215.67|2.92939|13.3|15.0|0.|1|1|1|-300.0|300.0|4772.06378216|4781.6240839|HI\n' \
+    'test.fits|1025.7222|2.92939|13.3|15.0|0.|1|1|1|-300.0|300.0|4026.43131868|4034.49782829|HI\n'
+    # test with comment
+    abscomp.comment = 'Something'
+    s = abscomp.repr_joevp('test.fits')
+    assert s == 'test.fits|1215.67|2.92939|13.3|15.0|0.|1|1|1|-300.0|300.0|4772.06378216|4781.6240839|HI# Something\n' \
+    'test.fits|1025.7222|2.92939|13.3|15.0|0.|1|1|1|-300.0|300.0|4026.43131868|4034.49782829|HI# Something\n'
+
+
+def test_complist_to_joevp():
+    # will write a file in directory ./files/
+    abscomp, HIlines = mk_comp('HI', b=15*u.km/u.s, use_rand=False)
+    comp_list = [abscomp, abscomp]
+    ltiu.complist_to_joevp(comp_list, 'test.fits', data_path('test_joevp_repr.joevp'))
+    # now read the output and compare to reference
+    compare_two_files(data_path('test_joevp_repr.joevp'), data_path('test_joevp_repr_reference.joevp'))
+    # now add attribute to comp and compare again
+    abscomp.attrib['b'] = 15*u.km/u.s
+    ltiu.complist_to_joevp(comp_list, 'test.fits', data_path('test_joevp_repr.joevp'))
+    compare_two_files(data_path('test_joevp_repr.joevp'), data_path('test_joevp_repr_reference.joevp'))
+
+
 def test_get_wvobs_chunks():
     abscomp, HIlines = mk_comp('HI', zcomp=0., vlim=[0,10]*u.km/u.s)
     wvobs_chunks = ltiu.get_wvobs_chunks(abscomp)
```

Comment:
Do you mean the user of IGMGuesses or the user of joebvp?


      From: Nicolas Tejos <notifications@github.com>
 To: linetools/linetools <linetools@noreply.github.com> 
Cc: jnburchett <jnburchett@yahoo.com>; Assign <assign@noreply.github.com>
 Sent: Tuesday, August 9, 2016 9:26 AM
 Subject: Re: [linetools/linetools] Joevp repr (#278)
   
In linetools/isgm/tests/test_use_abscomp.py:> +    'test.fits|1025.7222|2.92939|13.3|15.0|0.|1|1|1|-300.0|300.0|4026.43131868|4034.49782829|HI# Something\n'
> +
> +
> +def test_complist_to_joevp():
> +    # will write a file in directory ./files/
> +    abscomp, HIlines = mk_comp('HI', b=15*u.km/u.s, use_rand=False)
> +    comp_list = [abscomp, abscomp]
> +    ltiu.complist_to_joevp(comp_list, 'test.fits', data_path('test_joevp_repr.joevp'))
> +    # now read the output and compare to reference
> +    compare_two_files(data_path('test_joevp_repr.joevp'), data_path('test_joevp_repr_reference.joevp'))
> +    # now add attribute to comp and compare again
> +    abscomp.attrib['b'] = 15*u.km/u.s
> +    ltiu.complist_to_joevp(comp_list, 'test.fits', data_path('test_joevp_repr.joevp'))
> +    compare_two_files(data_path('test_joevp_repr.joevp'), data_path('test_joevp_repr_reference.joevp'))
> +
> +
>  def test_get_wvobs_chunks():
>      abscomp, HIlines = mk_comp('HI', zcomp=0., vlim=[0,10]*u.km/u.s)
>      wvobs_chunks = ltiu.get_wvobs_chunks(abscomp)
I have finally decided that the way to handle this, will be at the AbsComponent level, i.e. it is the user's responsibility to define AbsComponent with AbsLines that a given spectrum covers. In the case of IGMGuesses, a new PR will address this.—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub, or mute the thread.